app/jobs/asset_cleanup_job.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class AssetCleanupJob < ApplicationJob
+  def perform
+    active_revision_ids = Edition
+      .where(current: true)
+      .or(Edition.where(live: true))
+      .pluck(:revision_id)
+
+    active_file_attachment_asset_ids = FileAttachment::Revision
+      .joins(:revisions, blob_revision: [:asset])
+      .where("revisions.id" => active_revision_ids)
+      .pluck("file_attachment_assets.id")
+
+    active_image_asset_ids = Image::Revision
+      .joins(:revisions, blob_revision: [:assets])
+      .where("revisions.id" => active_revision_ids)
+      .pluck("image_assets.id")
+
+    FileAttachment::Asset
+      .where.not(id: active_file_attachment_asset_ids)
+      .where.not(state: "absent")
+      .each { |asset| delete_asset(asset) }
+
+    Image::Asset
+      .where.not(id: active_image_asset_ids)
+      .where.not(state: "absent")
+      .each { |asset| delete_asset(asset) }
+  end
```

Comment:
I'm not too sure about all the id extracting and passing around. Unbounded that can crash PostgreSQL once the number gets sufficiently high (I think maybe 2^16) and is potentially quite inefficient - this problem caused a lot of confusion when it broke Email Alert API.

I have a proposal that could help us avoid that, by effectively moving all the pre-extractions to be coverd

```
asset_exists_for_active_editions = Edition.where(current: %w[true live])
                                                                     .joins(revision: { image_revisions: :blob_revision })
                                                                     .where("image_assets.blob_revision_id = image_blob_revisions.id")
                                                                     .arel
                                                                     .exists

Image::Asset
  .where.not(state: :absent)
  .where.not(asset_exists_for_active_editions)
  .find_each { |asset| delete_asset(asset) }
```

This effectively moves the pre-loading of information into a subquery.